### DISCUSSION TITLE
Redirect??

### DISCUSSION THREAD

**Discussion by ismelferrer** [fa-fifi/keycloak_wrapper#44] [Q&A]
How can i handle the callback to my apk??
After the successfull login stay in the keycloak login page and don't go to the apk.....
&&&
What means this error "[keycloak_wrapper] No refresh token found."??

**Comment by fa-fifi** [comment]
"No refresh token found" is not an error, it just a log message. You probably received that if the user login for the first time.
To handle callback to ur app, just follow what my documentation said. If you stuck anywhere, you can ask here.

**Reply by ismelferrer** [reply]
I literally copy the example and don't went to the app, stay in the login page...even if i close the login page (after success login) the APK don't take the login data

**Reply by fa-fifi** [reply]
There's probably an issue with your Keycloak configuration. Have you change the valid redirect url inside your Keycloak admin console?

**Reply by jaderdev** [reply]
Hi, i'm having a similar issue! But i can login and be redirected back to the deep link by keycloak! But when the app receives de call it breaks because of http instead of https!

I/InsetsSourceConsumer(17349): applyRequestedVisibilityToControl: visible=false, type=ime, host=com.example.demo/com.example.demo.MainActivity
E/AndroidRuntime(17349): FATAL EXCEPTION: AsyncTask #1
E/AndroidRuntime(17349): Process: com.example.demo, PID: 17349
E/AndroidRuntime(17349): java.lang.RuntimeException: An error occurred while executing doInBackground()
E/AndroidRuntime(17349): 	at android.os.AsyncTask$4.done(AsyncTask.java:415)
E/AndroidRuntime(17349): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
E/AndroidRuntime(17349): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
E/AndroidRuntime(17349): 	at java.util.concurrent.FutureTask.run(FutureTask.java:269)
E/AndroidRuntime(17349): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
E/AndroidRuntime(17349): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
E/AndroidRuntime(17349): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
E/AndroidRuntime(17349): 	at java.lang.Thread.run(Thread.java:1012)
E/AndroidRuntime(17349): Caused by: java.lang.IllegalArgumentException: only https connections are permitted
E/AndroidRuntime(17349): 	at net.openid.appauth.Preconditions.checkArgument(Preconditions.java:116)
E/AndroidRuntime(17349): 	at net.openid.appauth.connectivity.DefaultConnectionBuilder.openConnection(DefaultConnectionBuilder.java:51)
E/AndroidRuntime(17349): 	at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:609)
E/AndroidRuntime(17349): 	at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:579)
E/AndroidRuntime(17349): 	at android.os.AsyncTask$3.call(AsyncTask.java:394)
E/AndroidRuntime(17349): 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
E/AndroidRuntime(17349): 	... 4 more


![Captura de Tela 2024-10-24 às 03 13 54](https://github.com/user-attachments/assets/7793803b-de08-4d8e-8d6e-7e99f419c459)

My keycloak.

The deeplink not works!

**Reply by zlatkoristevski** [reply]
> I literally copy the example and don't went to the app, stay in the login page...even if i close the login page (after success login) the APK don't take the login data

Hi, 

Did you find a solution about app stays in the login page?

Im facing the same issue. I followed the documentation and I think everything is set.

This is my keycloack config 
![image-111](https://github.com/user-attachments/assets/247d5600-f100-4843-96c4-ad6a4b684b24)

I've also included this into Info.plist

```
<key>CFBundleURLTypes</key>
	<array>
		<dict>
			<key>CFBundleTypeRole</key>
			<string>Editor</string>
			<key>CFBundleURLSchemes</key>
			<array>
				<string>com.example.he******er</string>
			</array>
		</dict>
	</array>
```

**Reply by fa-fifi** [reply]
@zlatkoristevski Try to remove `android:taskAffinity=""` from your AndroidManifest.xml file.

**Reply by zlatkoristevski** [reply]
> @zlatkoristevski Try to remove `android:taskAffinity=""` from your AndroidManifest.xml file.

The thing is that I'm trying only on iOS. Using Macbook, and trying only on ios emulator, iPhone 15max on ios 17.5. The problem still persist. Also, the flutter project is brand new with no dependencies installed except this one. 

Do you have any Keycloack test account to try out with your credentials just to see if by any chance the problem in the keycloack?

**Reply by fa-fifi** [reply]
Can you show me a video recording of the login process? Is there any error displayed inside the console log?

**Reply by zlatkoristevski** [reply]
> Can you show me a video recording of the login process? Is there any error displayed inside the console log?


https://github.com/user-attachments/assets/5151de06-3629-4068-b92f-b92c65cf232b


Here it is. No errors in the console.

In the debug tab this is only thing:

```
Launching lib/main.dart on iPhone 15 Pro Max in debug mode...
Xcode build done.                                           27.3s
Connecting to VM Service at ws://127.0.0.1:53662/GV32AQQTZwQ=/ws
Connected to the VM Service.
[keycloak_wrapper] No refresh token found.
```

**Reply by fa-fifi** [reply]
Sorry for the late reply. Quite busy at the moment. I'll run the example app on the iOS simulator by tonight and see if it's working or not. If anything, I'll update you.

**Reply by zlatkoristevski** [reply]
Did you manage to reproduce it?

**Reply by fa-fifi** [reply]
Sorry for the late reply. I've tried to replicate the issue using iOS simulator and iPhone, it's working fine. My current guess is maybe your Keycloak configuration is causing the issue. Need to research more about it.

**Reply by zlatkoristevski** [reply]
thanks for trying out for me. Could you possibly share your keylcloack credentials for me to try out or maybe share some tips regarding keycloack setup, except those that are in the documentation?

**Reply by fa-fifi** [reply]
Sorry but i can't share them since it is quite confidential.
